### DISCUSSION TITLE
Material texture change not working anymore

### DISCUSSION THREAD

**Discussion by plutorkhan** [google/model-viewer#3790] [Q&A]
### Description

The material change which i did through scene graph is not working anymor. When changing color the spa cabinet color should be changed as well by applying jpg texture to the material. It was working for months and no change has been made to the code, suddenly it stopped working.

Code: 
**Html**:  `<div style="width: 300px; margin-top: 50px;" id="controls">
       
       
       
       
       
       
       
       
       
       <input style="cursor:pointer;border-radius: 50%;
       background-color: #081430; width: 50px; border-width: 0px; height: 50px;" type="button" name="govde.jpg" class="colorPicker">
       <input style="cursor:pointer;border-radius: 50%;
       background-color: #888C8D; width: 50px; border-width: 0px; height: 50px;" type="button" name="govde2.jpg" class="colorPicker">
       <input style="cursor:pointer;border-radius: 50%;
       background-color: #3c1321; width: 50px; border-width: 0px; height: 50px;" type="button" name="govde3.jpg" class="colorPicker">
       
   </div>
`

**Javascript:**
<script type="module">
  const modelViewerTexture = document.querySelector("model-viewer");
  
 customElements.get("model-viewer").minimumRenderScale = 1;

  modelViewerTexture.addEventListener("scene-graph-ready", (ev) => {

      let material = modelViewerTexture.model.materials[17];
      let water = modelViewerTexture.model.materials[15];
      
       document.querySelectorAll('.colorPicker').forEach(e => {
          e.addEventListener('click', (event) => {
          material.pbrMetallicRoughness.baseColorTexture.texture.source.setURI(event.target.name);
      });
       })
     
      let checkbox = document.querySelector('input[type="checkbox"]');
      document.querySelectorAll('#switchWater').forEach(e => {
          e.addEventListener('change', (event) => {
            if (checkbox.checked) {
              water.pbrMetallicRoughness.setBaseColorFactor([1, 1, 1, 1]);
              modelViewerTexture.setAttribute("ios-src", "dre1600-2cl-water.usdz");
    console.log('Checked');
  } else {
    // do that
    water.pbrMetallicRoughness.setBaseColorFactor([1, 1, 1, 0]);
    modelViewerTexture.setAttribute("ios-src", "dre1600-2cl-dry.usdz");
  }
      });
       })

      document.querySelector("#controls").addEventListener('beforexrselect', (ev) => {
          ev.preventDefault();
      });
  })
</script>
#### Live Demo
https://imajinn.live/models/foxspa/dre-1600-2cl/index.html

### Version
https://unpkg.com/@google/model-viewer/dist/model-viewer.min.js
- model-viewer: vX.X.X

### Browser Affected

- [x ] Chrome, version: xx.x.xxxx.xx
- [ x] Edge
- [x ] Firefox
- [ ] IE
- [x ] Safari

### OS

- [ x] Android
- [x ] iOS
- [ ] Linux
- [x ] MacOS
- [x ] Windows

### AR

- [ x] WebXR
- [ ] SceneViewer
- [ ] QuickLook

**Comment by plutorkhan** [comment]
I couldn't put html in description. Though you can easily check the code in the link from chrome dev tools

**Comment by elalish** [comment]
It sounds like you should probably pin the version of MV you're using, until you can update to `createTexture`, see https://github.com/google/model-viewer/discussions/3788

**Comment by elalish** [comment]
Also, the real problem was an accidental push of a pre-release of v2.0 to unpkg, which has since been corrected: #3780.

**Reply by plutorkhan** [reply]
Could you kindly tell which version particularly supports scene graph api?

**Reply by elalish** [reply]
The scene graph API is still supported, only `setURI` was deprecated in favor of `createTexture` and `setTexture`, since they work more reliably and predictably. If you need `setURI` for now, pin to v1.12.
